src/shared/bootspec.c

```diff
@@ -895,6 +997,11 @@ static int boot_entries_find_unified(
                 if (r < 0)
                         continue;
 
+                /* look for .efi.extra.d */
+                r = boot_entries_find_unified_local_addons(config, full, de->d_name, config->entries + config->n_entries);
```

Comment:
hmm, please pass the dir fd through here, ad operate relative to that instead of using the full path again